src/lparser.c

```diff
@@ -846,14 +847,21 @@ void ykifyCode(lua_State *L, Proto *f, int num_insts) {
       loc_pc = pc - GETARG_Bx(i) + 2 - 1;
     } else {
       f->yklocs[pc] = yk_location_null();
-      continue;
+      goto cleanup;
     }
     YkLocation loc = yk_location_new();
 #if YKLUA_DEBUG_STRS
-    char *loc_dstr = luaG_ykdebug_str(f, loc_pc);
+    loc_dstr = luaG_ykdebug_str(f, loc_pc);
     yk_location_set_debug_str(&loc, loc_dstr);
 #endif
     f->yklocs[loc_pc] = loc;
+
+cleanup:
+#ifdef YKLUA_DEBUG_STRS
+    free(dstr);
+    if (loc_dstr)
+      free(dstr);
```

Comment:
potential double free here :) `s/dstr/loc_dstr/`

Reply:
Done in 309f58e.